record.go

```diff
@@ -178,3 +172,149 @@ func (d StructuredData) Bytes() []byte {
+func (c OpenCDCConverter) Name() string                                   { return "opencdc" }
+func (c OpenCDCConverter) Convert(r Record) (any, error)                  { return r, nil }
+
+type DebeziumConverter struct {
+	SchemaName string
+}
+
+func (c DebeziumConverter) Configure(opt map[string]string) (Converter, error) {
+	// allow user to configure the schema name (needed to make the output record
+	// play nicely with Kafka Connect connectors)
+	c.SchemaName = opt["debezium.schema.name"]
+	return c, nil
+}
+func (c DebeziumConverter) Name() string { return "debezium" }
+func (c DebeziumConverter) Convert(r Record) (any, error) {
+	before, err := c.getStructuredData(r.Payload.Before)
+	if err != nil {
+		return nil, err
+	}
+	after, err := c.getStructuredData(r.Payload.After)
+	if err != nil {
+		return nil, err
+	}
+
+	// we ignore the error, if the timestamp is not there milliseconds will be 0 which is fine
+	var readAtMillis int64
+	if readAt, err := r.Metadata.GetReadAt(); err == nil {
+		readAtMillis = readAt.UnixMilli()
+	}
+
+	dbz := kafkaconnect.DebeziumPayload{
+		Before:          before,
+		After:           after,
+		Source:          r.Metadata,
+		Op:              c.getDebeziumOp(r.Operation),
+		TimestampMillis: readAtMillis,
+		Transaction:     nil,
+	}
+
+	e := dbz.ToEnvelope()
+	e.Schema.Name = c.SchemaName
+	return e, nil
+}
+
+func (c DebeziumConverter) getStructuredData(d Data) (StructuredData, error) {
+	switch d := d.(type) {
+	case nil:
+		return nil, nil
+	case StructuredData:
+		return d, nil
+	case RawData:
+		if len(d) == 0 {
+			return nil, nil
+		}
+		return c.tryConvertRawDataToStructuredData(d)
+	default:
+		return nil, fmt.Errorf("unknown data type: %T", d)
+	}
+}
+func (c DebeziumConverter) tryConvertRawDataToStructuredData(d RawData) (StructuredData, error) {
+	// We have raw data, we need structured data.
+	// We can do our best and try to convert it if RawData is carrying raw JSON.
+	var sd StructuredData
+	err := json.Unmarshal(bytes.TrimSpace(d), &sd)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert RawData to StructuredData: %w", err)
+	}
+	return sd, nil
+}
+func (c DebeziumConverter) getDebeziumOp(o Operation) kafkaconnect.DebeziumOp {
+	switch o {
+	case OperationCreate:
+		return kafkaconnect.DebeziumOpCreate
+	case OperationUpdate:
+		return kafkaconnect.DebeziumOpUpdate
+	case OperationDelete:
+		return kafkaconnect.DebeziumOpDelete
+	case OperationSnapshot:
+		return kafkaconnect.DebeziumOpRead
+	}
+	return "" // invalid operation
+}
+
+type Encoder interface {
+	Name() string
+	Configure(options map[string]string) (Encoder, error)
+	Encode(r any) ([]byte, error)
+}
+
+var defaultEncoder = JSONEncoder{}
+
+type JSONEncoder struct{}
+
+func (e JSONEncoder) Configure(map[string]string) (Encoder, error) { return e, nil }
+func (e JSONEncoder) Name() string                                 { return "json" }
```

Comment:
is it better to have a one-line method if it's this simple? I'd prefer we format them across more lines tbh, my eyes don't like this 👀

Reply:
I see these methods as boilerplate code that need to be there just to fulfil the `Encoder` interface, so I didn't want to waste space with them. You read them once and forget about them. The interesting stuff is happening in `Encode` so I left that method formatted like a normal method, even though it could also fit in one line. That way the formatting guides you to the important part of this struct.

I've seen this formatting in multiple places, here's [one example](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/net/net.go;l=570-583). That said, it is a matter of taste, and not a rule or even a convention, so if I didn't convince you and you feel strongly about this we can reformat it 😉